### PR TITLE
ADD: Adding add_filter function to radar.py.

### PR DESCRIPTION
**Pull request by zssherman**
This function applies a filter to fields and either replaces or adds new filtered fields.

Note, adding tests to this PR, just initial review.

Closes, #763

### CODE REVIEW

File: pyart/core/radar.py [hunk 1]
```diff
@@ -788,6 +788,52 @@ def add_field(self, field_name, dic, replace_existing=False):
         self.fields[field_name] = dic
         return
 
+    def add_filter(self, gatefilter, replace_existing=False,
+                   include_fields=None):
+        """
+        Updates the radar object with an applied gatefilter provided
```
Comment (mgrover1) [inline pyart/core/radar.py:794]: Can you add more here, stating that it masks values?

**Review by mgrover1 — COMMENTED** [review]
One minor comment - otherwise, this looks great!

File: pyart/core/radar.py (hunk 1 above)
Comment (zssherman) [reply pyart/core/radar.py:794]: Sounds good!

**Comment by zssherman** [comment]
@mgrover1 Added unit tests as well now for review.

**Pull request merged**